client/src/index.tsx

```diff
@@ -22,12 +22,13 @@ if (!container) {
 // TODO: When we have TS types fo `docData` this would become
 // something like `(window as any).__data__ as DocData`.
 const docData = (window as any).__data__;
+const pageNotFound = (window as any).__pageNotFound__;
```

Comment:
This is critical so that when the client renders, it will know to NOT try to XHR fetch the `./index.json`. 
By putting clues like this into the global `window` scope, that's how we can communicate between server-sider rendered pages and client-side rendered pages.